lib/db_connection.ex

```diff
@@ -1330,8 +1332,8 @@ defmodule DBConnection do
   @doc """
   Declare a cursor.
 
-  Returns `{:ok, cursor}` on success or `{:error, exception}` if there was
-  an error.
+  Returns `{:ok, cursor}` or `{:ok, query, cursor}` on success or
```

Comment:
Do we have both return types for necessity or backwards compatibility? If backwards compatibility, it would be nice to document it as so.

Reply:
In postgrex PR both are returned. The new 3-tuple is only returned when the query result types change but we could always return it. The reason to return a new query is because we are able to survive type changes cases if parameter types are castable (from old to new) at the database and the result type can be anything. I think it will make sense for consistency to always return a query and leave the 2-tuple for backwards compatibility. However I'm not sure yet if we should encourage drivers to always return 3-tuple yet because haven't updated mariaex/ecto.